src/sidebar/util/groups.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+/**
+ * Combine groups from multiple api calls together to form a unique list of groups.
+ * Add a listGroup property to each group identifying it as a "mygroup" or a "featured" group.
+ *
+ * @param {Group[]} groups
+ */
+function combineGroups(results) {
+  const myGroups = results[0];
+  let featuredGroups = results[1];
+  let groups = [];
+
+  const worldGroup = featuredGroups.find(g => g.id === '__world__');
+  if (worldGroup) {
+    myGroups.unshift(worldGroup);
+  }
+
+  const myGroupIds = myGroups.map(g => g.id);
+  featuredGroups = featuredGroups.filter(g => !myGroupIds.includes(g.id));
+
+  // Set the groupList that each group belongs to.
+  featuredGroups.forEach(group => group.groupList = 'featured');
+  myGroups.forEach(group => group.groupList = 'mygroups');
```

Comment:
This is entirely a UI concern but there has to be some way of knowning which group list the groups should show up in or membership and non-membership at this point. Source makes me think it's pointing back to the API call that it came from but that's not really the case here. I wanted to be as explicit as possible so it's quite obvious it's which list it should show up in in the UI. Perhaps hasMembership or something but there's also the possibility that in the future we may have another groupList so I was trying to keep our options open there.

Reply:
I like the name of groupsList because it's easy to make a simple and explicit filter for it in the UI. Aka: for mygroups just filter the list by group.groupsList == 'mygroups'. I don't want to give the impression I'm not open to other names, I just want to make sure we choose a name that is explicit and `source` seems a bit cryptic to me.

Reply:
> Perhaps hasMembership or something but there's also the possibility that in the future we may have another groupList so I was trying to keep our options open there.

I wouldn't worry about the future here - we can always change the code when requirements change. A `hasMembership` boolean (I'd suggest `isMember` because we're interested in whether the current user _is a member_, rather than whether the group _has a membership_) captures the current distinction ... except for the Public group.

> Source makes me think it's pointing back to the API call that it came from but that's not really the case here.

Well, with the exception of "Public", every group in "myGroups" was present in the `/api/profile/groups` response. I think that probably suggests to me that the way we're currently planning to display "Public", as if it is a group that the user is a member of, is something we might want to reconsider.